force-app/main/default/classes/CON_PreferredPhone_TEST.cls

```diff
@@ -194,87 +208,137 @@ private class CON_PreferredPhone_TEST {
             Database.executeBatch(batch);
         Test.stopTest();
 
-        Contact newContAfterList = [SELECT Id, FirstName, LastName, Name, PreferredPhone__c FROM Contact];
-                
-        System.assertEquals('Other Phone', newContAfterList.PreferredPhone__c);
+        System.assertEquals('Other Phone', getCurrentContact().PreferredPhone__c);
        
     }   
     
-    /*  Validation when there are more than one phone fields but not Preferred Phone */
+    /* Validation when there are more than one phone fields but not Preferred Phone 
+     * Test Scenario: T-3903371
+     */
     @isTest 
-    static void testErrorPrefBlank() {
+    static void testErrorPrefPhoneAmbiguous() {
         enablePreferredPhone();
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            WorkPhone__c  = '111111111',
+            OtherPhone = '0000000000'
+        );
 
-        try {
-            Contact newCont = new Contact(
-                FirstName = 'JohnnyTest1',
-                LastName = 'JohnnyTest',
-                WorkPhone__c  = '111111111',
-                OtherPhone = '0000000000'
-            );
-            
+        try {    
             Test.startTest();
             insert newCont;
             Test.stopTest();
-            
+            System.assert(false, 'Multiple phones without preference specified should error out - when enforced');
         } catch(Exception error) {
              System.assert(error.getMessage().contains(Label.PreferredPhoneRequiredError));
+      }
+
+      disablePreferredPhone();
+      insert newCont;     
+      System.assert(true, 'Multiple phones without preference specified should not error out - when unenforced');
 
-      } 
-        
     } 
    
-    /* Validation when Preferred Phone is selected but no other Phone Fields have a value */
+    /* Ensure contact edits validate against missing specififed preferred phone(home phone) 
+    *  Test Scenario: T-3903370
+    */
     @isTest 
-    static void testPrefPhoneNotBlank() {
+    static void testcontactUpdatePrefPhoneNotBlankUponSettingUpdate() {
         enablePreferredPhone();
 
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            OtherPhone = '0000000000'
+        );        
+        insert newCont;
+
+        Hierarchy_Settings__c settings = enablePreferredPhone();
+        settings.Preferred_Phone_Selection__c = 'homePhone';
+        UTIL_CustomSettingsFacade.getSettingsForTests(settings);
+
+        newCont.PreferredPhone__c = 'Home phone';
         try {
-            Contact newCont = new Contact(
-                FirstName = 'JohnnyTest1',
-                LastName = 'JohnnyTest',
-                PreferredPhone__c  = 'Work'
-            );
+            Test.startTest();
+            update newCont;
+            Test.stopTest();
             
```

Comment:
The test should assert and error out in case if there was no exception during update